File: PROJETO_5 - WESLEY SANTOS/src/FuncionarioAdministrativo.py
from Usuario import Usuario

class FuncionarioAdministrativo(Usuario):
    def __init__(self, nome="", email="", departamento="", cargo=""):
        """
        Construtor padrão e parametrizado para a classe FuncionarioAdministrativo.
        """
        super().__init__(nome, email, "Funcionario")
        self.departamento = departamento
        self.cargo = cargo

    def gerar_relatorio(self):
        """
        Sobrescreve o método abstrato da classe base para exibir informações do funcionário administrativo.
        """
        print(f"Nome: {self.nome}\nDepartamento: {self.departamento}\nCargo: {self.cargo}")

File: FuncionarioAdministrativo.cpp
#include "FuncionarioAdministrativo.h"
#include <iostream>

FuncionarioAdministrativo::FuncionarioAdministrativo()
    : Usuario(), departamento(""), cargo("") {}

FuncionarioAdministrativo::FuncionarioAdministrativo(const std::string& nome, const std::string& email,
                                                   const std::string& departamento, const std::string& cargo)
    : Usuario(nome, email, "Funcionario Administrativo"), departamento(departamento), cargo(cargo) {}

void FuncionarioAdministrativo::gerarRelatorio() const {
    std::cout << "Funcionário Administrativo: " << nome << "\nDepartamento: " << departamento << "\nCargo: " << cargo << "\n" << std::endl;
}
